File: Java/프로그래머스/1/12901. 2016년/2016년.java
class Solution {
    public String solution(int a, int b) {
        int month[] = {31, 29, 31, 30, 31, 30, 31, 31, 30, 31, 30, 31};
        String[] week ={"THU","FRI","SAT","SUN", "MON", "TUE", "WED"};
        int sum = 0;
        for(int i=0; i<a-1; i++){
            sum+=month[i];
        }
        sum+=b;
        sum%=7;
        return week[sum];
    }
}
//1 3 5 7 8 10 12 -> 31
//4 6 9 11 -> 30
//2-> 29

File: Kotlin/프로그래머스/1/12901. 2016년/2016년.kt
class Solution {
    fun solution(a: Int, b: Int): String {
        var m = intArrayOf(0,31,29,31,30,31,30,31,31,30,31,30,31)
        var day = arrayOf("FRI","SAT","SUN","MON","TUE","WED","THU")
        return day[(m.slice(0..(a-1)).sum()+b-1)%7]
    }
}